packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +100,21 @@ export function* waitForGethConnectivity() {
   }
 }
 
+export async function waitForGethConnectivityAsync() {
+  let connected = isGethConnectedSelector(store.getState())
+  if (connected) {
+    return
+  }
+
+  while (true) {
```

Comment:
Can you add some number of retries similar to the waitForContractKit function to prevent getting stuck in this loop? https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/web3/contracts.ts#L126

Reply:
I considered doing that but opted to mirror the design after the generator function I was copying here: https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/geth/saga.ts#L80

I think it's probably a good idea to have a max number of retries but I wasn't sure if I was missing context for why we had the `while (true)` condition.